pkg/rpc/context.go

```diff
@@ -665,6 +665,24 @@ func NewContext(ctx context.Context, opts ContextOptions) *Context {
 	// a connection class. Only calls going over an actual gRPC connection will
 	// use that interceptor.
 
+	rpcCtx.Stopper.RunAsyncTask(ctx, "reconnect peers", func(ctx context.Context) {
```

Comment:
Also, and maybe this is making perfect the enemy of good, sort of sad to have a single goroutine that's always there and ticking but can then get stuck on an individual node.

We could try to introduce, on `peer`, an async probing-based circuit breaker (util/circuit) and make its probe loop responsible for reconnecting. This would be beneficial in its own right, since it would (later, and not related to your work directly) allow us to phase out the circuit breakers we have at the `nodedialer` level.